### PR TITLE
feat: Toasts

### PR DESCRIPTION
**Pull request by Shivani-1524**
Added Toasts and badge for icons on navbar and added state routing.
The preview works great after reloading it, incase you are facing issues.

Auth details:
email: test@email.com
password: test

files to review:
ToastProvider.js
ToastReducer.js
CartProduct.js
WishlistCard.js
ProductCard.js
Navbar.js
ArrivalCard.js
CategoryCard.js
CommonUtilities.js

### CODE REVIEW

**Review by Naman-Saxena1 — APPROVED** [review]
Code quality, syantx, variable naming looks good.
Was not able to find the reason why cart & wishlist management worked only after page releod of logged in user 👍

File: src/Reducers/ToastReducer.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const initToastState = {
+    toastList: []
+}
+const toastReducer = (state, { type, payload }) => {
+    console.log(state)
+    console.log(type, payload)
```
Comment (yaswanthmyneni) [inline src/Reducers/ToastReducer.js]: changes: remove console logs when pushing code to github

**Review by yaswanthmyneni — APPROVED** [review]
Toast implement is great, everything looks good to merge.

File: src/Pages/CartPage/CartComponents/CartProduct.js [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
-        const foundId = findProductInList(cartProduct._id, wishlistState)
-        if (!foundId) {
-            newWishlist = await addToWishlist(cartProduct)
-        } else {
-            console.log("product already in cart. update qty in cart page.")
-            newWishlist = wishlistState
-        }
-        wishlistDispatch({ type: 'UPDATE_WISHLIST', payload: newWishlist })
+        const { updatedList, toastMsg } = await handleAddToList({ addProductFn: addToWishlist, listState: wishlistState, productDetails: cartProduct })
+        console.log('tyty', updatedList, toastMsg)
```
Comment (gauravsinhaweb) [inline src/Pages/CartPage/CartComponents/CartProduct.js]: This console log is not required ig, you can remove it.

File: src/Pages/HomePage/HomeComponents/CategoryCard.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+import { Link } from 'react-router-dom'
 
-const CategoryCard = ({ img, caption }) => {
+const CategoryCard = ({ img, caption, filterCatKey }) => {
     return (
         <div className="category-preview">
-            <a href="./Products/products.html">
+            <Link to="/shop" state={{ filterEnabled: filterCatKey }}>
                 <div className="round-img-container">
                     <img className="round-img img-resp" src={img} alt="Hoodie" />
```
Comment (gauravsinhaweb) [inline src/Pages/HomePage/HomeComponents/CategoryCard.js]: keep alt valueDynamic as well!

File: src/Pages/HomePage/HomeComponents/NewArrivalCard.js [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
             <div className="grid-txt-overlay pos-abs">
                 <p className="rg-title light-txt center-txt">{title}</p>
-                <a href="./Products/products.html" >
+                <Link to="/shop" state={{ gameEnabled: filterGameKey, filterEnabled: filterCatKey }}>
                     <button className="btn subtle-btn mg-t-10 center-items">Explore <i className="fa fa-solid fa-arrow-right"></i>
                     </button>
-                </a>
+                </Link>
             </div>
             <img className="img-resp" src={imgSrc} alt="Xiao Genshin Poster" />
```
Comment (gauravsinhaweb) [inline src/Pages/HomePage/HomeComponents/NewArrivalCard.js]: keep alt value Dynamic!

File: src/Reducers/ToastReducer.js (hunk 1 above)
Comment (gauravsinhaweb) [inline src/Reducers/ToastReducer.js]: remove all the console logs if not required!

File: src/index.js [hunk 5]
```diff
@@ -17,7 +17,9 @@ ReactDOM.render(
         <FilterProvider>
           <WishlistProvider>
             <CartProvider>
-              <App />
+              <ToastProvider>
+                <App />
+              </ToastProvider>
             </CartProvider>
           </WishlistProvider>
         </FilterProvider>
```
Comment (gauravsinhaweb) [inline src/index.js:25]: context provider looks so clean!

**Review by gauravsinhaweb — APPROVED** [review]
1. context looks so clean and readable.
2.  Keep alt/ caption in IMG tag dynamic tag.
some minor changes I suggested, Good to merge after changes.

**Comment by Shivani-1524** [comment]
> Code quality, syantx, variable naming looks good. Was not able to find the reason why cart & wishlist management worked only after page releod of logged in user 👍

Thank You! and yeah i have no idea why :/

File: src/index.js (hunk 5 above)
Comment (Shivani-1524) [reply src/index.js:25]: thank you

File: src/Pages/HomePage/HomeComponents/NewArrivalCard.js (hunk 4 above)
Comment (Shivani-1524) [reply src/Pages/HomePage/HomeComponents/NewArrivalCard.js]: oh thanks for pointing it out. completely overlooked it.

File: src/Pages/HomePage/HomeComponents/CategoryCard.js (hunk 3 above)
Comment (Shivani-1524) [reply src/Pages/HomePage/HomeComponents/CategoryCard.js]: done

File: src/Pages/CartPage/CartComponents/CartProduct.js (hunk 2 above)
Comment (Shivani-1524) [reply src/Pages/CartPage/CartComponents/CartProduct.js]: done

**Comment by Shivani-1524** [comment]
> 1. context looks so clean and readable.
> 2. Keep alt/ caption in IMG tag dynamic tag.
>    some minor changes I suggested, Good to merge after changes.

Thank you gaurav

File: src/Reducers/ToastReducer.js (hunk 1 above)
Comment (Shivani-1524) [reply src/Reducers/ToastReducer.js]: done

**Comment by Shivani-1524** [comment]
> Toast implement is great, everything looks good to merge.

thank you :)

**Review by dipen2000 — COMMENTED** [review]
The UI is looking good , the toasts are looking good too.

**Pull request merged**